app/scripts/models/account.js

```diff
@@ -16,31 +16,41 @@ define([
   'lib/constants',
   'models/profile-image'
 ], function (Backbone, _, p, AuthErrors, ProfileClient, Constants, ProfileImage) {
-
   // Account attributes that can be persisted
   var PERSISTENT = {
-    uid: undefined,
+    accessToken: undefined,
     email: undefined,
+    grantedPermissions: undefined,
+    lastLogin: undefined,
+    needsOptedInToMarketingEmail: undefined,
+    profileImageId: undefined,
+    profileImageUrl: undefined,
     sessionToken: undefined,
     sessionTokenContext: undefined,
-    accessToken: undefined,
-    verified: undefined,
-    profileImageUrl: undefined,
-    profileImageId: undefined,
-    lastLogin: undefined,
-    grantedPermissions: undefined
+    uid: undefined,
+    verified: undefined
   };
 
   var DEFAULTS = _.extend({
-    password: undefined,
-    unwrapBKey: undefined,
+    customizeSync: undefined,
     keyFetchToken: undefined,
-    customizeSync: undefined
+    password: undefined,
+    unwrapBKey: undefined
   }, PERSISTENT);
 
   var ALLOWED_KEYS = Object.keys(DEFAULTS);
   var ALLOWED_PERSISTENT_KEYS = Object.keys(PERSISTENT);
 
+  function withMarketingEmailClient(method) {
+    return function () {
+      var self = this;
+      return self.getOAuthTokenFromSessionToken(self.get('sessionToken'), 'basket')
```

Comment:
> I've been waffling whether email marketing related functionality should live here and an email-marketing-client is
> passed in, or whether the email marketing client should be passed to the views and its functions accept an account. 

Tough call.

In isolation, I'd *slightly* prefer to have an "email marketing client" that takes the account to operate on as an argument, either in the constructor or in each method call.  But I have a strong default preference for keeping a low method count on each class.  People have criticized my style in the past for preferring "manager objects" a little to much...

But given that the account already handles profile information by exposing profile-manipulation methods that call through to an underlying profile client, I think you made the right call here by keeping the approach consistent.  We can consider factoring both of these out if the account object becomes too unweildly under the weight of all its special-purpose methods.

(It's not just about cognitive load either - one concrete downside of making these all methods on the Account, is that we must load the code for all of them even if the user never sees anything to do with their profile or email marketing prefs.  Something to think about in the future if we ever want to chunk up our downloads a little more).